doc/feature/restAPI.md

```diff
@@ -0,0 +1,129 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented,
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+GET /registry returns the URIs for all registered objects, together with the
+supported REST operation (GET or PUT). Each objet can be queried for its schema
+using a GET operation on the URI of that object, suffixed by /schema.
+
+- Example:
+  - Get all URIs of registered objects:
+    * Request: ``` GET http://hostname:port/registry ```
+    * Response:
+```
+    {
+      { "namespace/foo": ["GET", "PUT"] },
+      { "namespace/bar": ["PUT"] }
+    }
+```
+  - Get JSON schema for a specific object:
+    * Request: ```GET http://hostname:port/namespace/foo/schema```
+    * Response:```{ "foo": { "name": string, "age": integer } }```
+
+## Examples
+Create an HTTP server and register serializables:
+```
+  void setupHTTPServer( const int argc, char** argv )
+  {
+    // Create HTTP server
+    _httpServer = zeq::http::Server::parse( argc, argv );
+    if( !_httpServer )
+      return;
+
+    // Register C++ objects
+    subscribers.push_back( _httpServer );
+    _httpServer.subscribe( camera, LUT, vrParams );
+    _httpServer.register( camera, LUT, vrParams, frame );
+  }
+```
+Query API:
+   * Request: ```GET http://hostname:port/zerobuf/render/camera/schema```
+   * Response: ```{ "camera" : { "origin": "string", "lookat": "up" } }```
+
+## Implementation
+- Add a getSchema public method to Servus::serializable
+- Make sure getSchema is implemented by to the code generator and returns the
+  content of the .fbs file.
+- _registry_ and _schema_ are both keywords that correspond to GET requests.
+  They both need to be handled by the _processGet method of the
+  zeq::http::Server implementation.
+- Add a getRegistry method to the zeq::http::Server implementation that returns
+  the contents of _registrations and _subscriptions containers in the form of
+  a JSON list a tuples defined by a URI and a list of REST operations (currently
+  GET and PUT).
+- This mechanism should be transparent to the user of the class, and should be
+  handled internally to the zeq::http::Server. The schema can be requested at
+  any time on invocation of the getSchema method on the registered object.
+
+## Issues
+### API
+Three options are being discussed regarding the implementation of the API:
```

Comment:
Issue is more a list of questions that need to be resolved. In this case we should document which of the three options was chosen and why.